<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Function reference â€¢ genomicMateSelectR</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Function reference" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-index">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">genomicMateSelectR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index">

    <colgroup>
      
      <col class="alias" />
      <col class="title" />
    </colgroup>

    <tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-genomic-mate-selection-functions" class="hasAnchor"><a href="#section-genomic-mate-selection-functions" class="anchor"></a>Genomic mate selection functions</h2>
          <p class="section-desc"><p>Top level functions to predict the performance of individual genotypes and the usefulness of potential crosses.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="runGenomicPredictions.html">runGenomicPredictions()</a></code> </p>
        </td>
        <td><p>Run genomic predictions</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="predictCrosses.html">predictCrosses()</a></code> </p>
        </td>
        <td><p>Predict crosses</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-cross-prediction-functions" class="hasAnchor"><a href="#section-cross-prediction-functions" class="anchor"></a>Cross prediction functions</h2>
          <p class="section-desc"><p>Behind-the-scenes functions that predict cross means and variances.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="predCrossVars.html">predCrossVars()</a></code> </p>
        </td>
        <td><p>Predict cross variances</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="predCrossMeans.html">predCrossMeans()</a></code> </p>
        </td>
        <td><p>Predict cross means</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calcGameticLD.html">calcGameticLD()</a></code> </p>
        </td>
        <td><p>Calculate the gametic LD matrix for a single parent</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calcCrossLD.html">calcCrossLD()</a></code> </p>
        </td>
        <td><p>Calculate the cross LD matrix based on both parents</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-cross-validation-functions" class="hasAnchor"><a href="#section-cross-validation-functions" class="anchor"></a>Cross-validation functions</h2>
          <p class="section-desc"><p>Functions for automatically conducting cross-validation-based estimation of genomic prediction accuracy.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="runCrossVal.html">runCrossVal()</a></code> </p>
        </td>
        <td><p>Run k-fold cross-validation</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="runParentWiseCrossVal.html">runParentWiseCrossVal()</a></code> </p>
        </td>
        <td><p>Run parent-wise cross-validation</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-nextgen-cassava-genomic-selection" class="hasAnchor"><a href="#section-nextgen-cassava-genomic-selection" class="anchor"></a>NextGen Cassava genomic selection</h2>
          <p class="section-desc"><p>Functions used in the standardized steps of the NextGen Cassava genomic selection breeding pipeline</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      <tr>
        <th colspan="2">
          <h3 id="section-cassavabase-phenotype-data-pipeline-functions" class="hasAnchor"><a href="#section-cassavabase-phenotype-data-pipeline-functions" class="anchor"></a>Cassavabase phenotype data pipeline functions</h3>
          <p class="section-desc"><p>Functions for processing of cleaning and otherwise preparing field data.</p></p>
        </th>
      </tr>
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="calcPropMissing.html">calcPropMissing()</a></code> </p>
        </td>
        <td><p>Calcluate the proportion missing data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="curateTrialOneTrait.html">curateTrialOneTrait()</a></code> </p>
        </td>
        <td><p>Fit model and remove outliers from one trial for one trait</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="curateTrialsByTrait.html">curateTrialsByTrait()</a></code> </p>
        </td>
        <td><p>Fits mixed-models to each trial and all traits, removing outliers</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="detectExptDesigns.html">detectExptDesigns()</a></code> </p>
        </td>
        <td><p>"Detect" experimental designs</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="fitMultiTrialModel.html">fitMultiTrialModel()</a></code> </p>
        </td>
        <td><p>Fit model and remove outliers from a multi-trial dataset</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="makeTrialTypeVar.html">makeTrialTypeVar()</a></code> </p>
        </td>
        <td><p>Add TrialType classifier</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="nestByTrials.html">nestByTrials()</a></code> </p>
        </td>
        <td><p>Nest data.frame by trials</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="nestDesignsDetectedByTraits.html">nestDesignsDetectedByTraits()</a></code> </p>
        </td>
        <td><p>Set-up for a by-trait analysis downstream</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="nestForMultiTrialAnalysis.html">nestForMultiTrialAnalysis()</a></code> </p>
        </td>
        <td><p>Restructure BLUPs for downstream analysis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="nestTrialsByTrait.html">nestTrialsByTrait()</a></code> </p>
        </td>
        <td><p>Nest trials by trait</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="readDBdata.html">readDBdata()</a></code> </p>
        </td>
        <td><p>Read database downloads into R</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="renameAndSelectCols.html">renameAndSelectCols()</a></code> </p>
        </td>
        <td><p>Rename columns and remove everything unecessary</p></td>
      </tr>
    </tbody><tbody>
      
      <tr>
        <th colspan="2">
          <h3 id="section-imputation-and-informatics" class="hasAnchor"><a href="#section-imputation-and-informatics" class="anchor"></a>Imputation and Informatics</h3>
          <p class="section-desc"><p>Functions wrapping around standard command line bioinformatics tools, used for setting up and running imputation and post-imputation filtering on the genotypes needing prediction.</p></p>
        </th>
      </tr>
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="convertDart2vcf.html">convertDart2vcf()</a></code> </p>
        </td>
        <td><p>Convert DArTseqLD reports to VCF</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="convertVCFtoDosage.html">convertVCFtoDosage()</a></code> </p>
        </td>
        <td><p>Convert a VCF file to dosage matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="createGenomewideDosage.html">createGenomewideDosage()</a></code> </p>
        </td>
        <td><p>Combine multiple dosage matrix to a single genome-wide dosage</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="filter_positions.html">filter_positions()</a></code> </p>
        </td>
        <td><p>Subset a VCF file based on provided positions</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mergeVCFs.html">mergeVCFs()</a></code> </p>
        </td>
        <td><p>Merge two VCFs into one</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="postImputeFilter.html">postImputeFilter()</a></code> </p>
        </td>
        <td><p>Apply quality filters after imputation by Beagle5.0</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="postImputeFilterBeagle4pt1.html">postImputeFilterBeagle4pt1()</a></code> </p>
        </td>
        <td><p>Apply quality filters after imputation by Beagle4.1</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="runBeagle4pt1GL.html">runBeagle4pt1GL()</a></code> </p>
        </td>
        <td><p>Run Beagle4.1 to impute a target VCF with a reference VCF</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="runBeagle5.html">runBeagle5()</a></code> </p>
        </td>
        <td><p>Run Beagle5 to impute a target VCF with a reference VCF</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="splitVCFbyChr.html">splitVCFbyChr()</a></code> </p>
        </td>
        <td><p>Split a genome-wide VCF into separate chromosome-wise VCFs</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-helper-functions" class="hasAnchor"><a href="#section-helper-functions" class="anchor"></a>Helper functions</h2>
          <p class="section-desc"><p>Support functions big and small!</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="backsolveSNPeff.html">backsolveSNPeff()</a></code> </p>
        </td>
        <td><p>Compute ridge-regression equivalent marker effects from GBLUP results ("Backsolve SNP effects")</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="centerDosage.html">centerDosage()</a></code> </p>
        </td>
        <td><p>Centers dosage matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="crosses2predict.html">crosses2predict()</a></code> </p>
        </td>
        <td><p>Make a data.frame of pairwise crosses between a set parents</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dose2domDev.html">dose2domDev()</a></code> </p>
        </td>
        <td><p>Converts a dosage matrix into a matrix of centered "classical"-coded dominance deviations.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dose2domDevGenotypic.html">dose2domDevGenotypic()</a></code> </p>
        </td>
        <td><p>Converts a dosage matrix into a matrix of centered "genotypic"-coded dominance deviations.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="effectsArray2list.html">effectsArray2list()</a></code> </p>
        </td>
        <td><p>Converts an array of posterior samples of multi-trait marker effects to a named list (one for each trait).</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="genmap2recombfreq.html">genmap2recombfreq()</a></code> </p>
        </td>
        <td><p>Create a matrix of pairwise recombination frequencies from a genetic map</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="genoVarCovarMatFunc.html">genoVarCovarMatFunc()</a></code> </p>
        </td>
        <td><p>Compute the LD matrix from dosages</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="getAF.html">getAF()</a></code> </p>
        </td>
        <td><p>Compute allele frequencies</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="getMAF.html">getMAF()</a></code> </p>
        </td>
        <td><p>Compute minor allele frequencies</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="getPropHom.html">getPropHom()</a></code> </p>
        </td>
        <td><p>Compute the per-individual proportion homozygous</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="intensity.html">intensity()</a></code> </p>
        </td>
        <td><p>Standardized selection intensity</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="kinship.html">kinship()</a></code> </p>
        </td>
        <td><p>Construct several types of kinship matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="maf_filter.html">maf_filter()</a></code> </p>
        </td>
        <td><p>Filter a dosage matrix by a minor allele frequency</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="quadform.html">quadform()</a></code> </p>
        </td>
        <td><p>Quadratic matrix form</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="remove_invariant.html">remove_invariant()</a></code> </p>
        </td>
        <td><p>Remove invariant SNPs from dosage matrix</p></td>
      </tr>
    </tbody>
    </table>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Marnin Wolfe.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


