<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Run k-fold cross-validation — runCrossVal • genomicMateSelectR</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Run k-fold cross-validation — runCrossVal" />
<meta property="og:description" content="Assess the accuracy of predicted previously unobserved genotypes
(individuals) based on the available training data. Runs k-fold
cross-validation for potentially multiple traits and optionally computing
prediction accuracy on user-specified selection index. Three models are
enabled: additive-only (&quot;A&quot;), additive-plus-dominance (&quot;AD&quot;) and a
directional-dominance model that incorporates a genome-wide homozygosity
effect (&quot;DirDom&quot;). The union of all genotypes scored for all traits is broken
into k-folds a user specified number of times. Subsequently each train-test
pair is predicted for each trait and accuracies are computed." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">genomicMateSelectR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Run k-fold cross-validation</h1>
    
    <div class="hidden name"><code>runCrossVal.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Assess the accuracy of predicted previously unobserved genotypes
(individuals) based on the available training data. Runs k-fold
cross-validation for potentially multiple traits and optionally computing
prediction accuracy on user-specified selection index. Three models are
enabled: additive-only ("A"), additive-plus-dominance ("AD") and a
directional-dominance model that incorporates a genome-wide homozygosity
effect ("DirDom"). The union of all genotypes scored for all traits is broken
into k-folds a user specified number of times. Subsequently each train-test
pair is predicted for each trait and accuracies are computed.</p>
    </div>

    <pre class="usage"><span class='fu'>runCrossVal</span><span class='op'>(</span>
  <span class='va'>blups</span>,
  <span class='va'>modelType</span>,
  <span class='va'>selInd</span>,
  SIwts <span class='op'>=</span> <span class='cn'>NULL</span>,
  <span class='va'>grms</span>,
  dosages <span class='op'>=</span> <span class='cn'>NULL</span>,
  <span class='va'>nrepeats</span>,
  <span class='va'>nfolds</span>,
  ncores <span class='op'>=</span> <span class='fl'>1</span>,
  nBLASthreads <span class='op'>=</span> <span class='cn'>NULL</span>,
  gid <span class='op'>=</span> <span class='st'>"GID"</span>,
  seed <span class='op'>=</span> <span class='cn'>NULL</span>,
  <span class='va'>...</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>blups</th>
      <td><p>nested data.frame with list-column "TrainingData" containing
BLUPs. Each element of "TrainingData" list, is data.frame with de-regressed
BLUPs, BLUPs and weights (WT) for training and test.</p></td>
    </tr>
    <tr>
      <th>modelType</th>
      <td><p>string, "A", "AD", "DirDom". modelType="A": additive-only,
GEBVS modelType="AD": the "classic" add-dom model, GEBVS+GEDDs = GETGVs
modelType="DirDom": the "genotypic" add-dom model with prop. homozygous fit
as a fixed-effect, to estimate a genome-wide inbreeding effect. obtains
add-dom effects, computes allele sub effects (\(\alpha = a + d(q-p)\))
incorporates into GEBV and GETGV. "DirDom" requires dosages</p></td>
    </tr>
    <tr>
      <th>selInd</th>
      <td><p>logical, TRUE/FALSE, selection index accuracy estimates,
requires input weights via <code>SIwts</code></p></td>
    </tr>
    <tr>
      <th>SIwts</th>
      <td><p>required if <code>selInd=FALSE</code>, named vector of selection index
weights, names match the "Trait" variable in <code>blups</code></p></td>
    </tr>
    <tr>
      <th>grms</th>
      <td><p>list of GRMs where each element is named either A, D, or, AD.
Matrices supplied must match required by A, AD and ADE models. For ADE
grms=list(A=A,D=D)</p></td>
    </tr>
    <tr>
      <th>dosages</th>
      <td><p>dosage matrix. required only for modelType=="DirDom". Assumes
SNPs coded 0, 1, 2. Nind rows x Nsnp cols, numeric matrix, with rownames
and colnames to indicate SNP/ind ID</p></td>
    </tr>
    <tr>
      <th>nrepeats</th>
      <td><p>number of repeats</p></td>
    </tr>
    <tr>
      <th>nfolds</th>
      <td><p>number of folds</p></td>
    </tr>
    <tr>
      <th>ncores</th>
      <td><p>number of cores, parallelizes across repeat-folds</p></td>
    </tr>
    <tr>
      <th>nBLASthreads</th>
      <td><p>number of cores for each worker to use for multi-thread
BLAS</p></td>
    </tr>
    <tr>
      <th>gid</th>
      <td><p>string variable name used for genotype ID's/ in e.g. <code>blups</code>
(default="GID")</p></td>
    </tr>
    <tr>
      <th>seed</th>
      <td><p>numeric, use seed to achieve reproducibile train-test folds.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Returns tidy results in a tibble with accuracy estimates for each rep-fold in a list-column "accuracyEstOut".</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p>Other CrossVal: 
<code><a href='runParentWiseCrossVal.html'>runParentWiseCrossVal</a>()</code></p></div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Marnin Wolfe.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


